### PR TITLE
Add a manual SingleFileExporter

### PR DESCRIPTION
**Pull request by theduke**
Allows manually exporting a set of types to a single file.

### CODE REVIEW

**Comment by Mattchine** [comment]
I like this. Thank you.
Ps. I think this PR break when enable `format` feature.

**Comment by aaronleopold** [comment]
Happy to help wherever needed get this merged, assuming it is within the scope of the project? Single file exporting (e.g. index.ts) is something I personally would really like to see, but understand if it isn't something the maintainer wants 🙂

**Comment by joaoantoniocardoso** [comment]
I'm also missing this feature - I hacked my way out to do the same thing in a custom build.rs, but it would be way better to have this feature merged/available.

**Comment by escritorio-gustavo** [comment]
Updated tests to expect `type` instead of `interface´

**Comment by escritorio-gustavo** [comment]
@NyxCode I'm not familiar with this part of the codebase, I think you should have a look

File: ts-rs/src/export.rs [hunk 1]
```diff
@@ -161,3 +161,93 @@ where
...
+    }
+
+    pub fn and<T: TS + 'static>(mut self) -> Self {
+        self.add_type::<T>();
+        self
+    }
+
+    /// Finalize the export.
+    ///
+    /// Returns the generated typescritp code on success, or an error if any
```
Comment (escritorio-gustavo) [inline ts-rs/src/export.rs:230]: Typo in "typescript"

File: ts-rs/src/export.rs [hunk 2]
```diff
@@ -161,3 +161,93 @@ where
...
+    buffer: String,
+}
+
+impl SingleFileExporter {
+    pub fn new(with_codegen_warning: bool) -> Self {
+        let buffer = if with_codegen_warning {
+            NOTE.to_string()
+        } else {
+            String::new()
+        };
```
Comment (escritorio-gustavo) [inline ts-rs/src/export.rs:195]: @NyxCode should the codegen warning be optional?

**Review by escritorio-gustavo — APPROVED** [review]

**Comment by gustavo-shigueo** [comment]
Apart from me cleaning up merge conflicts, this has been stale for over a year now, so I'll be closing this issue

**Pull request closed without merging**